### PR TITLE
Rename labels to fix missing nnsFunction descriptions

### PR DESCRIPTION
**Pull request by mstrasinskis**
# Motivation

Some `nnsFunction` enum keys have been updated in [ic-js](https://github.com/dfinity/ic-js/pull/695), but the corresponding keys in the NNS-dapp governance labels were not updated. This discrepancy leads to missing type field for the renamed types in the NNS-dapp. In this PR, we synchronize the `nnsFunction` label keys to match the `ic-js` enum.

# Changes

- Rename `UpdateSubnetReplicaVersion` to `DeployGuestosToAllSubnetNodes`.
- Rename `UpdateElectedReplicaVersions` to `ReviseElectedGuestosVersions`.
- Rename `AddApiBoundaryNode` to `AddApiBoundaryNodes`.
- Drive-by: update `getNnsFunctionKey` test name.

# Tests

- Added a test to verify that there are labels for all provided `nnsFunction` values.
- Tested locally to confirm that the `DeployGuestosToAllSubnetNodes` proposal now correctly displays the type field after the changes (which was previously missing).

# Todos

- [ ] Add entry to changelog (if necessary).
Not necessary.

### CODE REVIEW

File: frontend/src/lib/i18n/en.governance.json [hunk 1]
```diff
@@ -151,12 +151,14 @@
     "UpdateAllowedPrincipals": "Update Allowed Principals",
     "RetireReplicaVersion": "Retire Replica Version",
     "InsertSnsWasmUpgradePathEntries": "Insert SNS-WASM Upgrade Path Entries",
-    "UpdateElectedReplicaVersions": "Revise Elected GuestOS Versions",
+    "ReviseElectedGuestosVersions": "Revise Elected GuestOS Versions",
     "BitcoinSetConfig": "Set Bitcoin Config",
     "UpdateElectedHostosVersions": "Update Elected HostOS Versions",
     "UpdateNodesHostosVersion": "Update Nodes HostOS Version",
-    "AddApiBoundaryNode": "Add API Boundary Nodes",
+    "HardResetNnsRootToVersion": "HardResetNnsRootToVersion",
```
Comment (dskloetd) [inline frontend/src/lib/i18n/en.governance.json:158]: This string seems strange. Shouldn't there at least be some spaces?

File: frontend/src/lib/i18n/en.governance.json [hunk 2]
```diff
@@ -151,12 +151,14 @@
...
+    "AddApiBoundaryNodes": "Add API Boundary Nodes",
     "RemoveApiBoundaryNodes": "Remove API Boundary Nodes",
+    "UpdateApiBoundaryNodeDomain": "UpdateApiBoundaryNodeDomain",
```
Comment (dskloetd) [inline frontend/src/lib/i18n/en.governance.json:161]: Maybe change to `"Deprecated - not used"`, or something?

File: frontend/src/lib/i18n/en.governance.json [hunk 3]
```diff
@@ -204,14 +206,16 @@
...
     "BitcoinSetConfig": "A proposal to set the configuration of the underlying Bitcoin Canister that powers the Bitcoin API. The configuration includes whether or not the Bitcoin Canister should sync new blocks from the network, whether the API is enabled, the fees to charge, etc.",
     "UpdateElectedHostosVersions": "A proposal to update currently elected HostOS versions, by electing a new version, and/or unelecting multiple unused versions. The version to elect (often identified by the hash of the installation image) is added to the registry. By itself, this proposal does not effect any upgrade. The specified versions to unelect are removed from the registry. This ensures that the HostOS cannot upgrade to these versions anymore.",
     "UpdateNodesHostosVersion": "Update the HostOS version running on a given list of nodes. The proposal changes the HostOS version that is used on the specified nodes. The version must be contained in the list of HostOS versions.",
-    "AddApiBoundaryNode": "A proposal to add a set of new API Boundary Nodes using unassigned nodes.",
+    "HardResetNnsRootToVersion": "HardResetNnsRootToVersion",
```
Comment (dskloetd) [inline frontend/src/lib/i18n/en.governance.json:213]: same

File: frontend/src/lib/i18n/en.governance.json [hunk 4]
```diff
@@ -204,14 +206,16 @@
...
+    "AddApiBoundaryNodes": "A proposal to add a set of new API Boundary Nodes using unassigned nodes.",
     "RemoveApiBoundaryNodes": "A proposal to remove a set of API Boundary Nodes, which will designate them as unassigned nodes.",
+    "UpdateApiBoundaryNodeDomain": "UpdateApiBoundaryNodeDomain",
```
Comment (dskloetd) [inline frontend/src/lib/i18n/en.governance.json:216]: same

File: frontend/src/lib/i18n/en.governance.json [hunk 5]
```diff
@@ -204,14 +206,16 @@
...
     "UpdateApiBoundaryNodesVersion": "A proposal to update the version of a set of API Boundary Nodes.",
-    "DeployGuestosToSomeApiBoundaryNodes": "A proposal to update the version of a set of API Boundary Nodes.",
+    "DeployGuestosToSomeApiBoundaryNodes": "A proposal (DeployGuestosToSomeApiBoundaryNodes) to update the version of a set of API Boundary Nodes.",
```
Comment (dskloetd) [inline frontend/src/lib/i18n/en.governance.json:218]: Why did you change this?

Comment (mstrasinskis) [reply frontend/src/lib/i18n/en.governance.json:218]: The new unit test checks the uniqueness of nnsFunction labels. This was necessary to make the deprecated `DeployGuestosToSomeApiBoundaryNodes` description unique, as it currently shares the same description as `DeployGuestosToAllUnassignedNodes`.

Comment (dskloetd) [reply frontend/src/lib/i18n/en.governance.json:218]: How about treating this as an exception *in the test*, rather than changing the product for the sake of a test?

File: frontend/src/lib/i18n/en.governance.json (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/lib/i18n/en.governance.json]: Removed.

File: frontend/src/lib/i18n/en.governance.json (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/lib/i18n/en.governance.json]: Removed.

File: frontend/src/lib/i18n/en.governance.json (hunk 3 above)
Comment (mstrasinskis) [reply frontend/src/lib/i18n/en.governance.json]: Has been removed.

File: frontend/src/lib/i18n/en.governance.json (hunk 4 above)
Comment (mstrasinskis) [reply frontend/src/lib/i18n/en.governance.json]: Moved to the IGNORED_IDS in the test.

File: frontend/src/lib/i18n/en.governance.json (hunk 5 above)
Comment (mstrasinskis) [reply frontend/src/lib/i18n/en.governance.json]: Makes sense. The PR is much cleaner now.

File: frontend/src/tests/lib/utils/proposals.utils.spec.ts [hunk 6]
```diff
@@ -620,6 +621,45 @@ describe("proposals-utils", () => {
         typeDescription
       );
     });
+
+    it("should provide labels for all function IDs", () => {
+      const IGNORED_NNS_FUNCTION_IDS = [
+        NnsFunction.HardResetNnsRootToVersion,
```
Comment (dskloetd) [inline frontend/src/tests/lib/utils/proposals.utils.spec.ts:627]: If you know, it could be nice to have a comment on each of them why it's ignored.

**Review by dskloetd — APPROVED** [review]
Thanks!

File: frontend/src/tests/lib/utils/proposals.utils.spec.ts (hunk 6 above)
Comment (mstrasinskis) [reply frontend/src/tests/lib/utils/proposals.utils.spec.ts:627]: Updated.
